### ISSUE TITLE
Error when using Minindn with Nlsr security=True: minindn/apps/nlsr.py: KeyError: 'workDir'

### ISSUE THREAD

**Issue by italovalcy** [named-data/mini-ndn#44]
When you enable the security mode of Minindn/Nlsr (argument --security) the startup process gives the following error (either with or without --work-dir arg)
```
~/mini-ndn$ sudo python examples/nlsr/pingall.py --security --work-dir /tmp/minindn
Using topology file /usr/local/etc/mini-ndn/default-topology.conf
*** Creating network
*** Adding controller
*** Adding hosts:
a b c d
*** Adding switches:

*** Adding links:
(10ms delay) (10ms delay) (a, b) (10ms delay) (10ms delay) (a, c) (10ms delay) (10ms delay) (b, d)
*** Configuring hosts
a b c d
Dummy key chain patch is installed in ndn-cxx. Security will be disabled.
*** Starting controller
c0
*** Starting 0 switches

Traceback (most recent call last):
  File "examples/nlsr/pingall.py", line 50, in <module>
    logLevel='ndn.*=TRACE:nlsr.*=TRACE')
  File "/home/minindn/mini-ndn/minindn/apps/app_manager.py", line 33, in __init__
    self.startOnNode(host, **appParams)
  File "/home/minindn/mini-ndn/minindn/apps/app_manager.py", line 38, in startOnNode
    app = self.cls(host, **appParams)
  File "/home/minindn/mini-ndn/minindn/apps/nlsr.py", line 93, in __init__
    self.createKeysAndCertificates()
  File "/home/minindn/mini-ndn/minindn/apps/nlsr.py", line 113, in createKeysAndCertificates
    securityDir = '{}/security'.format(self.parameters['workDir'])
KeyError: 'workDir'
```

The problem seems to be related to the way Nlsr app expects to access the Minindn working dir to create the root keys/certificate (minindn/apps/nlsr.py:113). The current code expects workDir to be part of the node parameter (Line 50: self.parameters = self.node.params['params'] and Line 113: securityDir = '{}/security'.format(self.parameters['workDir'])), however the workDir is actually an global parameter of Minindn and should be available for any app that may need it. Even for distributed experiments, the app copies some files from the Minindn workDir. The nodes will have the homeDir (based on the workDir). Thus, maybe it is better to change Nlsr app to use workDir from Minindn instead from node's attributes.

I would say the same about the resultsDir, but since there is no app using it now, it gives no problem now.

**Comment by italovalcy** [comment]
Using the proposed patch, I was able to enable security mode:
```
~/mini-ndn$ sudo python examples/nlsr/pingall.py --security --work-dir /tmp/minindn --nPings 20
Using topology file /usr/local/etc/mini-ndn/default-topology.conf
*** Creating network
*** Adding controller
*** Adding hosts:
a b c d
*** Adding switches:

*** Adding links:
(10ms delay) (10ms delay) (a, b) (10ms delay) (10ms delay) (a, c) (10ms delay) (10ms delay) (b, d)
*** Configuring hosts
a b c d
ERROR: No default identity
*** Starting controller
c0
*** Starting 0 switches

ndnsec-keygen /ndn/
ndnsec-cert-dump -i /ndn/ > /tmp/minindn/security/root.cert
ndnsec-certgen -s /ndn/ -r /tmp/minindn/a/security/site.keys > /tmp/minindn/a/security/site.cert
ndnsec-certgen -s /ndn/ -r /tmp/minindn/b/security/site.keys > /tmp/minindn/b/security/site.cert
ndnsec-certgen -s /ndn/ -r /tmp/minindn/c/security/site.keys > /tmp/minindn/c/security/site.cert
ndnsec-certgen -s /ndn/ -r /tmp/minindn/d/security/site.keys > /tmp/minindn/d/security/site.cert
Waiting 60 seconds for convergence...
...done
NLSR has converged successfully.
Each node will ping 4 node(s)
Scheduling ping(s) from a to b
Scheduling ping(s) from a to c
Scheduling ping(s) from a to d
Scheduling ping(s) from b to c
Scheduling ping(s) from b to d
Scheduling ping(s) from b to a
Scheduling ping(s) from c to d
Scheduling ping(s) from c to a
Scheduling ping(s) from c to b
Scheduling ping(s) from d to a
Scheduling ping(s) from d to b
Scheduling ping(s) from d to c
*** Starting CLI:
mini-ndn>
*** Stopping 1 controllers
c0
*** Stopping 3 links
...
*** Stopping 0 switches

*** Stopping 4 hosts
a b c d
*** Done
~/mini-ndn$ grep -c content /tmp/minindn/*/ping-data/*.txt
/tmp/minindn/a/ping-data/b.txt:20
/tmp/minindn/a/ping-data/c.txt:20
/tmp/minindn/a/ping-data/d.txt:20
/tmp/minindn/b/ping-data/a.txt:20
/tmp/minindn/b/ping-data/c.txt:20
/tmp/minindn/b/ping-data/d.txt:20
/tmp/minindn/c/ping-data/a.txt:20
/tmp/minindn/c/ping-data/b.txt:20
/tmp/minindn/c/ping-data/d.txt:20
/tmp/minindn/d/ping-data/a.txt:20
/tmp/minindn/d/ping-data/b.txt:20
/tmp/minindn/d/ping-data/c.txt:20
```

**Comment by dulalsaurab** [comment]
Thank you for the patch. I think you are right. workDir and resultDir should be the class variable. 

Can you please submit your patch via gerrit, our master is under gerrit control, meaning the margin has to go via gerrit. Additionally, other people can/will also review your work and possibly provide feedback, if needed. 

if you haven't submitted ndn related patch to gerrit before, here is a document on how you can do:  https://gerrit.named-data.net/q/project:mini-ndn

**Comment by dulalsaurab** [comment]
Problem is fixed by [ccd85b1cf1efd1b45e3648b9f243afc93fb9419d](https://github.com/named-data/mini-ndn/commit/ccd85b1cf1efd1b45e3648b9f243afc93fb9419d)
